lib/govuk_index/popularity_worker.rb

```diff
@@ -4,41 +4,52 @@ class PopularityWorker < Indexer::BaseWorker
     QUEUE_NAME = "bulk".freeze
     sidekiq_options queue: QUEUE_NAME
 
-    def perform(records, destination_index)
-      actions = Index::ElasticsearchProcessor.new(client: GovukIndex::Client.new(timeout: BULK_INDEX_TIMEOUT, index_name: destination_index))
+    def perform(document_ids, index_name)
+      actions = Index::ElasticsearchProcessor.new(
+        client: GovukIndex::Client.new(timeout: BULK_INDEX_TIMEOUT, index_name:),
+      )
+
+      source_index = IndexFinder.by_name(index_name)
+
+      popularities = retrieve_popularities_for(index_name, document_ids)
+      document_ids.each do |document_id|
+        document = source_index.get_document_by_id(document_id)
+        unless document
+          puts "Skipping #{document_id} as it is not in the index"
```

Comment:
I think I probably made a bit of a janky suggestion of mine to use puts in the workers. I think instead we use `logger.warn`